var/spack/repos/builtin/packages/fastqc/package.py

```diff
@@ -42,21 +40,20 @@ class Fastqc(Package):
     patch('fastqc.patch', level=0)
 
     def install(self, spec, prefix):
-        mkpath(self.prefix.bin)
-        mkpath(self.prefix.lib)
-        copy_file('fastqc', self.prefix.bin)
+        mkdir(prefix.bin)
+        mkdir(prefix.lib)
+        install('fastqc', prefix.bin)
         for j in ['cisd-jhdf5.jar', 'jbzip2-0.9.jar', 'sam-1.103.jar']:
-            copy_file(j, self.prefix.lib)
+            install(j, prefix.lib)
         for d in ['Configuration', 'net', 'org', 'Templates', 'uk']:
-            copy_tree(d, join_path(self.prefix.lib, d))
+            install_tree(d, join_path(prefix.lib, d))
         chmod = which('chmod')
-        chmod('+x', join_path(self.prefix.bin, 'fastqc'))
+        chmod('+x', prefix.bin.fastqc)
 
     # In theory the 'run' dependency on 'jdk' above should take
```

Comment:
This seems like a problem that should have an issue attached to it, is there one?

Reply:
@citibeth reported a bug similar to this but I was never able to reproduce it. It may have been a bug once upon a time that was fixed.

Reply:
Do we know someone who uses this package and could test, whether this workaround is still necessary?

Reply:
You could see who wrote the package. Looking back, I think what happened was that they installed this package, loaded the module, but Java wasn't found. They need to manually load the Java module or tell Spack to recursively load modules for run dependencies.